component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpResponseProcessor.java

```diff
@@ -0,0 +1,117 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.util.stream.Collectors;
+
+/**
+ * Handles sending data to source listener.
+ */
+public class HttpResponseProcessor implements Runnable {
+    private static final Logger logger = LoggerFactory.getLogger(HttpResponseProcessor.class);
+    private HTTPCarbonMessage carbonMessage;
+    private SourceEventListener sourceEventListener;
+    private String sinkId;
+    private String[] trpProperties;
+
+    private String filePath;
+
+    HttpResponseProcessor(HTTPCarbonMessage cMessage, SourceEventListener sourceEventListener,
+                          String sinkId, String[] trpProperties) {
+        this.carbonMessage = cMessage;
+        this.sourceEventListener = sourceEventListener;
+        this.sinkId = sinkId;
+        this.trpProperties = trpProperties;
+    }
+    
+    @Override
+    public void run() {
+        int code = carbonMessage.getNettyHttpResponse().status().code() / 100;
+        boolean isDownloadableContent = (boolean) (carbonMessage.getProperty(HttpConstants.IS_DOWNLOADABLE_CONTENT));
+
+        if (isDownloadableContent && code == 2) {
+            filePath = carbonMessage.getProperty(HttpConstants.DOWNLOAD_PATH).toString();
+            String fileName = writeToTile(carbonMessage);
+            if (fileName != null) {
+                sourceEventListener.onEvent(fileName, trpProperties);
+            }
+        } else {
+            BufferedReader buf = new BufferedReader(
+                    new InputStreamReader(
+                            new HttpMessageDataStreamer(carbonMessage).getInputStream(), Charset.defaultCharset()));
+            try {
+                String payload = buf.lines().collect(Collectors.joining("\n"));
+                if (!payload.equals(HttpConstants.EMPTY_STRING)) {
+                    sourceEventListener.onEvent(payload, trpProperties);
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("Submitted Event :" + payload);
+                    }
+                } else {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("Empty payload event, hence dropping the event chunk in the source with sink" +
+                                ".id : " + sinkId);
+                    }
+                }
+            } finally {
+                try {
+                    buf.close();
+                } catch (IOException e) {
+                    logger.error("Error closing byte buffer.", e);
+                }
+            }
+        }
+    }
+
+    private String writeToTile(HTTPCarbonMessage carbonMessage) {
+        File file = new File(filePath);
+        try (
+                InputStream inputStream = new HttpMessageDataStreamer(carbonMessage).getInputStream();
+                OutputStream outputStream = new FileOutputStream(file);
+
```

Comment:
Above code is not formatted properly. Please check.

Reply:
Fixed with 8f6f25e